src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -283,5 +282,20 @@ public override Uri GetUri(string name)
         {
             return new Uri(_directory.Uri, name);
         }
+
+        public override async Task<bool> AreSyncronized(Uri sourceUri, Uri destinationUri)
+        {
+            var destination = new CloudBlockBlob(destinationUri);
```

Comment:
This assumes public blob right? I think you should be calling `GetBlockBlobReference` instead of the constructor. You can fix this easily-ish by accepting `sourceUri`'s `Storage` as a parameter.

Reply:
For destination we can do that. For source we cannot with the current design - Catalog2DNX works under the assumption that the source is public.

Reply:
Okay, sounds good.